samples/basic/blink_led/src/main.c

```diff
@@ -38,12 +44,19 @@ void main(void)
 		return;
 	}
 
-	/* Adjust max_period depending pwm capabilities */
-	pwm_get_cycles_per_sec(pwm_dev, PWM_CHANNEL, &cycles);
-	/* in very worst case, PWM has a 8 bit resolution and count up to 256 */
-	period = max_period = 256 * USEC_PER_SEC / cycles;
-	min_period = max_period / 64;
+	/* In case the default MAX_PERIOD value cannot be set for some PWM
+	 * hardware, try to decrease the value until it fits, but no further
+	 * than to the value of MIN_PERIOD muliplied by four (to allow the
+	 * sample to actually show some blinking with changing frequency).
+	 */
+	max_period = MAX_PERIOD;
```

Comment:
Maybe this sample should be just renamed/refactored, I think the goal we are trying to fix the wrong problem, the goal here is just demoing PWM usage, so we don't really care about the **period** but about the **duty cycle**. Maybe the sample should just vary the duty cycle from e.g. 10% to 100%  and 100% to 10% over one-second cycles.

Reply:
The purpose of this sample seems to be to show how PWM can be used for just blinking a LED, not to show PWM usage in general (the fade_led sample seems to better fit the latter). And this sample has been present in the Zephyr tree for three years, so perhaps people find it useful. The description of the sample promises people to see some blinking, but after 16d8ce519c9fe13a2a7bfd23552eafe93b422b49 they cannot see it on some platforms, so in its current form the sample is confusing. And I'm trying to fix only this.

Reply:
> The purpose of this sample seems to be to show how PWM can be used for just blinking a LED, not to show PWM usage in general.

Yes that the point, It's not a common usage for PWM, this will end with a lot of boards not supporting the promises because of their clock freq limitation. So maybe an explicit print about this limitation in case of failure finding min freq would be useful. Or maybe just blacklist the incompatible boards...

>  (the fade_led sample seems to better fit the latter)

true.

> so in its current form the sample is confusing. And I'm trying to fix only this.

Yes I understand, It's more general comment about this sample than on your patch.

Reply:
> So maybe an explicit print about this limitation in case of failure finding min freq would be useful.

Something like this?
```suggestion
	max_period = MAX_PERIOD;
	while (pwm_pin_set_usec(pwm_dev, PWM_CHANNEL,
				max_period, max_period / 2U)) {
		if (max_period > (4U * MIN_PERIOD)) {
			max_period /= 2U;
		} else {
			printk("This sample needs to set a period that is "
			       "not supported by the used PWM driver.");
			return;
		}
	}
```

Reply:
yes!

Reply:
Applied. In a slightly modified form, to better fit the preceding comment.